improver/nbhood/square_kernel.py

```diff
@@ -70,14 +71,25 @@ def __init__(
             neighbourhood_method:
                 Name of the neighbourhood method to use. Options: 'circular',
                 'square'.
```

Comment:
This doesn't make sense now. Why have the option to select a circular neighbourhood from within the square_kernel module? If you plan to remove this in a follow-up PR, then I'm happy for it to stay temporarily, but putting "deprecated" somewhere in this bit of the doc-string would be good.

Reply:
This is now a general neighbourhooding plugin which does support both square and circular neighbourhood processing. The whole plugin is in the wrong place right now, but I haven't moved any plugins to their final home yet, instead I've been consolidating code in a way that makes the diff most manageable. I'm planning to move all the plugins to the right place as a final PR once there is the smallest amount of code to move, and I'm hoping we can agree the end locations for everything upfront before I start moving things.

Is that ok?

Reply:
Yes, that's fine. You did say you were going to move things in the next PR. I'll approve this in a moment.